api/leaderboard.go

```diff
@@ -661,18 +641,12 @@ func (app *App) GetTopPercentage(ctx context.Context, req *api.GetTopPercentageR
 		if err != nil {
 			lg.Error("Enriching members failed.", zap.Error(err))
 
-			if !errors.Is(err, enriching.ErrNotConfigured) {
-				lg.Error("Enriching members failed.", zap.Error(err))
-				app.AddError()
-				return nil, status.Errorf(codes.Internal, "Unable to enrich members")
-			}
-
-			lg.Debug("Enrichment not configured.", zap.Error(err))
+			lg.Error("Enriching members failed.", zap.Error(err))
+			app.AddError()
+			return nil, status.Errorf(codes.Internal, "Unable to enrich members")
 		}
 
-		if result != nil {
-			members = result
-		}
+		members = result
```

Comment:
Instead of having this, what about changing line 640 to:

```go
members, err := app.Enricher.Enrich(ctx, tenantID[0], req.LeaderboardId, members)
```